src/hardware/mixer.cpp

```diff
@@ -3042,8 +3053,11 @@ static void init_mixer_dosbox_settings(Section_prop& sec_prop)
 
 	bool_prop = sec_prop.Add_bool("negotiate", OnlyAtStart, DefaultAllowNegotiate);
 	bool_prop->Set_help(
-	        "Let the system audio driver negotiate possibly better sample rate and blocksize\n"
-	        "settings (%s by default).");
+	        "Negotiate a possibly better sample rate setting that does not require resampling\n"
+	        "the emulator's sound output, and an optimal block size (%s by default).\n"
+	        "Disable it if you want to force a specific block size and sample rate (e.g., to\n"
+	        "capture audio at an emulated audio device's native rate), but that might incur\n"
+	        "additional resampling at the host OS audio level.");
```

Comment:
"additional resampling at the OS level" isn't quite accurate.  If you turn sample rate negotiate off and SDL cannot send the requested sample rate to the OS, it will do resampling inside SDL.

IMO this should be avoided at all cost because SDL's resampler is quite primitive.  I've looked at that code when it seg faulted on me.  That bug has since been fixed but I still would avoid using it.

I don't see a benefit to ever have sample rate negotiation turned off.  We have a Speex re sampler than can handle any arbitrate sample rate and it does a better job than SDL so just use that.

The worst case happens when a user requests some weird sample rate that their hardware/OS doesn't support.  We resample to that weird rate, hand it to SDL, and it just re-samples it a 2nd time before passing it to the OS.

Reply:
Okay, you're right @weirddan455. For some reason, it did not occur to me I can do that if the user set `rate` and the SDL-negotiated sample frame rate mismatch... 🤦🏻 That's some tunnel vision, eh? 😆

I kinda assumed the current SDL resampler is not as shit as it used to be, but yeah, it's best to take the resampling matter 100% out of SDL's hands, and it's easy to do so with Speex.

Thanks man! 😎 🤘🏻

Reply:
Actually, as discussed, this is not needed.